### ISSUE TITLE
nctoolbox makes MATLAB verbose?

### ISSUE THREAD

**Issue by lakshaynarula** [nctoolbox/nctoolbox#95]
Hi,

After I start nctoolbox in MATLAB, it seems like MATLAB starts printing all the code it executes on the command window until I restart MATLAB. I find that somewhat annoying. Is there any way to disable this behavior while starting nctoolbox?

I've tried multiple releases of nctoolbox and I see the same behavior on both Mac OS and Ubuntu.

Thanks,
Lakshay.

**Comment by hohonuuli** [comment]
That doesn't sound like anything nctoolbox is doing. In your startup.m, file do you have a line that looks like `echo on`?

**Comment by lakshaynarula** [comment]
No, but there are a bunch of `echo on`'s in the ncgeodataset.m file that might be causing this. If I call `echo off` after every call to ncgeodataset, it seems to solve the problem except it still prints "echo off" on the command window.

**Comment by rsignell-usgs** [comment]
@lakshaynarula , which version of nctoolbox are you using?

**Comment by lakshaynarula** [comment]
I am currently using 1.1.0 because the later versions were unable to read some of the GRIB files. This is the latest version that is able to read all files I have. The issue of echo is persistent in all later versions I have used too. I think it is because of the `echo on` commands in the ncgeodataset.m file.

**Comment by rsignell-usgs** [comment]
@lakshaynarula , okay, got it. 

@acrosby , this looks you implemented this:
https://github.com/nctoolbox/nctoolbox/commit/aa17e2382bfca4ebdc5d90a8b1f99b5e99f2eb63
did you just have a moment of dyslexia with `echo on` and `echo off`?   Shouldn't these be reversed?

**Comment by acrosby** [comment]
I suspect those were there for debugging, they can be removed.

On Mon, Jan 23, 2017, 06:40 Rich Signell <notifications@github.com> wrote:

@lakshaynarula <https://github.com/lakshaynarula> , okay, got it.

@acrosby <https://github.com/acrosby> , this looks you implemented this:
aa17e23
<https://github.com/nctoolbox/nctoolbox/commit/aa17e2382bfca4ebdc5d90a8b1f99b5e99f2eb63>
did you just have a moment of dyslexia with echo on and echo off? Shouldn't
these be reversed?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/nctoolbox/nctoolbox/issues/95#issuecomment-274466400>,
or mute the thread

**Comment by acrosby** [comment]
actually I remember now, I think I was trying to suppress statements that
were coming from in there that I couldn't control. either way the offs and
ons can be removed

**Comment by rsignell-usgs** [comment]
@lakshaynarula , do you have an example of a grib file that you could read with 1.1.0 but not 1.1.1?
